### PR TITLE
Make paypal blocks work in FSE & widget editor

### PR DESCRIPTION
**Pull request by dsas**
Partly addresses [calypso#61319](https://github.com/Automattic/wp-calypso/issues/61319) - it addresses the issue with the widget editor and FSE but not the customizer

#### Changes proposed in this Pull Request:

Paypal blocks didn't work in the FSE & widget editors, you could add and edit them, but they wouldn't actually display on the front-end unless you saved and then totally reloaded the page.

#### Testing instructions:

Happy path:
  1. Switch to a theme that supports widgets, e.g. Twenty Nineteen
  2. Go to Appearance > Widgets
  3. Add a new Pay with Paypal block, fill out the details and press Save
  4. Go to the front-end and notice that the block is displayed

Other places:
As above, but also check adding the block to
 * Posts (should continue to work)
 * FSE: Use Twenty TwentyTwo theme and go to Appearance->Edit whatsit (didn't work before, **should work now**)
 * Appearance -> Customizer -> Widgets (didn't work before, **still doesn't work now**)

The Customizer doesn't work on current production - it allows you to add a widget to the footer and save, but when you view the site, the widget isn't there. Reloading the page, or opening it in the Widget Editor gives a partially saved widget - no title, description or image image. After applying this patch, the experience is exactly the same - **there isn't a regression**.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/simple-payments/edit.js [hunk 1]
```diff
@@ -616,7 +616,9 @@ const mapSelectToProps = withSelect( ( select, props ) => {
 
 	return {
 		hasPublishAction: !! get( post, [ '_links', 'wp:action-publish' ] ),
-		isSaving: !! isSavingPost(),
+		isSaving: __experimentalGetDirtyEntityRecords().some( record =>
```
Comment (BogdanUngureanu) [inline projects/plugins/jetpack/extensions/blocks/simple-payments/edit.js:619]: nitpick: I think it's a common pattern to use an alias for _experimental functions when importing them.

Comment (dsas) [reply projects/plugins/jetpack/extensions/blocks/simple-payments/edit.js:619]: Are there any further conventions around experimental functions? Looking at discussions, this one has been labelled experimental for some time.

Comment (BogdanUngureanu) [reply projects/plugins/jetpack/extensions/blocks/simple-payments/edit.js:619]: No idea, maybe @Copons knows other?

**Comment by rcrdortiz** [comment]
Works as expected on Atomic.
On WPCom I'm unable to apply the patch (clean repo and up to date):
Captura de pantalla 2022-03-07 a las 13 01 33

**Comment by dsas** [comment]
If I sync up the changes manually then it is fine.

After speaking with Jeremy I added these changes to @BogdanUngureanu's D75148-code. This diff relies upon that being there, and it also fixes a bug that that diff would introduce, so it makes sense.

Therefore please test D75148-code (once built) and not the phab diff mentioned in the auto-comment

File: projects/plugins/jetpack/extensions/blocks/simple-payments/edit.js (hunk 1 above)
Comment (dsas) [reply projects/plugins/jetpack/extensions/blocks/simple-payments/edit.js]: I think he thinks it's fine.

**Comment by rcrdortiz** [comment]
Tests well on Simple sites too.

**Review by rcrdortiz — APPROVED** [review]

**Review by jeherve — APPROVED** [review]
This tests well. 🚢

**Comment by dsas** [comment]
Deployed r241654-wpcom

**Pull request merged**